### PR TITLE
adding envFrom for deployments

### LINKED ISSUE #87: support for envFrom for secrets
This change helps avoid having to enumerate all secrets and keep the management of the secrets isolated to the secrets config and not both the secret config and the chart.

https://v1-18.docs.kubernetes.io/docs/concepts/configuration/secret/#use-case-as-container-environment-variables

### PR DESCRIPTION
**Pull request by ralucas**
This adds the envFrom key for containers.  Resolves #87

### CODE REVIEW

**Review by yorinasub17 — COMMENTED** [review]
Thanks for your contribution! The code looks good to me, but I think it makes more sense to expose this using the existing `configMaps` and `secrets` inputs by introducing a new type key for `as`. E.g.,

```
configMaps:
  myconfig:
    as: envFrom
```

What do you think?

**Comment by ralucas** [comment]
@yorinasub17 Thanks for the review.  I've made the requested adjustments.  Let me know if I've missed something, etc.  Thanks!

File: charts/k8s-service/values.yaml [hunk 1]
```diff
@@ -371,11 +371,11 @@ additionalContainerEnv: {}
...
 # entry in the map represents a ConfigMap resource. The key refers to the name of the ConfigMap that should be exposed,
 # with the value specifying how to expose the ConfigMap. The value is also a map and has the following attributes:
-#   - as (enum[volume,environment,none]) (required)
+#   - as (enum[volume,environment,envFrom,none]) (required)
 #     : ConfigMaps can be exposed to Pods as a volume mount, or as environment variables. This attribute is a string
-#       enum that is expected to be either "volume" or "environment", specifying that the ConfigMap should be exposed as a
-#       mounted volume or via environment variables respectively. This attribute can also be set to "none", which
-#       disables the `ConfigMap` on the container.
+#       enum that is expected to be either "volume", "environment", or "envFrom", specifying that the ConfigMap should
+#       be exposed as a mounted volume, via environment variables, or loaded as environment variables respectively.
```
Comment (yorinasub17) [inline charts/k8s-service/values.yaml:377]:
```suggestion
#       be exposed as a mounted volume, via environment variables, or loaded in its entirety as environment variables respectively.
```

File: charts/k8s-service/values.yaml [hunk 2]
```diff
@@ -400,7 +400,8 @@ additionalContainerEnv: {}
 # https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/
 #
 # The following example exposes the ConfigMap `myconfig` as a volume mounted to `/etc/myconfig`, while it exposes the
-# ConfigMap `myotherconfig` as an environment variable.
+# ConfigMap `myotherconfig` as an environment variable.  Additionally, it automatically mounts `anotherconfig` as
```
Comment (yorinasub17) [inline charts/k8s-service/values.yaml:403]:
```suggestion
# ConfigMap `myotherconfig` as an environment variable.  Additionally, it automatically mounts all the keys of `anotherconfig` as
```

File: charts/k8s-service/values.yaml [hunk 3]
```diff
@@ -447,11 +450,11 @@ scratchPaths: {}
...
 # the map represents a Secret resource. The key refers to the name of the Secret that should be exposed, with the value
 # specifying how to expose the Secret. The value is also a map and has the following attributes:
-#   - as (enum[volume,environment,none]) (required)
+#   - as (enum[volume,environment,envFrom,none]) (required)
 #     : Secrets can be exposed to Pods as a volume mount, or as environment variables. This attribute is a string enum
-#       that is expected to be either "volume" or "environment", specifying that the Secret should be exposed as a mounted
-#       volume or via environment variables respectively. This attribute can also be set to "none", which disables the
-#       `Secret` on the container.
+#       that is expected to be either "volume", "environment", or "envFrom", specifying that the Secret should be
+#       exposed as a mounted volume, via environment variables, or loaded as environment variables respectively.
```
Comment (yorinasub17) [inline charts/k8s-service/values.yaml:456]:
```suggestion
#       exposed as a mounted volume, via environment variables, or loaded in its entirety as environment variables respectively.
```

**Review by yorinasub17 — COMMENTED** [review]
Updates LGTM! Just had a few minor docs changes and then it's good to go! I'll kick off a regression build now.

**Review by yorinasub17 — APPROVED** [review]
Build passed, so will merge this in! Thanks again for your contribution!

**Pull request merged**